src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -263,5 +268,46 @@ public string CreateAuthorizationHeader()
                 _authenticationScheme?.AuthorizationHeaderPrefix ?? TokenType,
                 AccessToken);
         }
+
+#if !NETSTANDARD1_3
+        /// <summary>
+        /// Returns a <See cref= "ClaimsPrincipal"/> object containing the claims extracted from the ID Token.
+        /// </summary>
+        public ClaimsPrincipal GetIdTokenClaims()
```

Comment:
this logic is the same as [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/src/client/Microsoft.Identity.Client/Internal/IdToken.cs#L9) shouldn't this go in the IdToken class and be exposed a different way?

Reply:
I think it's somewhat different. Internall, we deserialize the id token to an object called IDToken. Neha's deserializing the idtoken to a dictionary and creating a ClaimsPrincipal. 

There may a bit of code duplication in how we parse that id token string (i.e. the splitting) - possibly worth have a static method somewhere to reduce duplication.

Reply:
Agree, there is a small part of code that is duplicated. I will try to refactor it into a static method.